src/foam/nanos/notification/email/POP3EmailService.java

```diff
@@ -0,0 +1,289 @@
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeBodyPart;
+import javax.mail.internet.MimeMessage;
+import javax.mail.internet.MimeMultipart;
+import javax.mail.Part;
+import javax.mail.UIDFolder;
+
+import javax.activation.DataHandler;
+import javax.activation.DataSource;
+import javax.activation.FileDataSource;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.io.FileInputStream;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.BufferedOutputStream;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import com.sun.mail.pop3.POP3Folder;
+import com.sun.mail.imap.IMAPFolder;
+
+import java.lang.Object;
+
+public class POP3EmailService extends ContextAwareSupport implements POP3Email, NanoService
+{
+  public void fetch(String pop3Host, String storeType, String user, String password) {
+    DAO ticketDAO = (DAO) getX().get("ticketDAO");
+
+    try {
+      Properties properties = new Properties();
+      properties.setProperty("mail.store.protocol", "imaps");
+      // properties.put("mail.store.protocol", "pop3");
+      properties.put("mail.pop3.host", pop3Host);
+      properties.put("mail.pop3.port", "995");
+      properties.put("mail.pop3.starttls.enable", "true");
+      Session emailSession = Session.getDefaultInstance(properties);
+      emailSession.setDebug(true);
+
+      Store store = emailSession.getStore("imaps");
+
+      store.connect(pop3Host, user, password);
+      Folder emailFolder = store.getFolder("INBOX");
+      emailFolder.open(Folder.READ_ONLY);
+      Message[] messages = emailFolder.getMessages();
+              
+      IMAPFolder imapfolder = (IMAPFolder) emailFolder;
+      System.out.println("messages.length---" + messages.length);
+        
+      for (int i = 0; i < messages.length; i++) {
+        Message message = messages[i];
+        long emailId = imapfolder.getUID(message);
+        System.out.println("--------emailId--------");
+        // iterate through tickets from the ticketDAO, compare emailId on ticket, if its not there create new ticket
+        // by setting the ticket emailId to the emailId from the imapfolder.getUID() method.
+        // and putting to  TicketDAO
+
+        try{ 
+          ArraySink sink = (ArraySink) ticketDAO.where(EQ(Ticket.EMAIL_ID, emailId)).select(new ArraySink());
+          List ticketList = sink.getArray();
+          System.out.println(ticketList); 
+        }
+        catch ( Throwable e ) {
+         
+        }
+
+
+        System.out.println(emailId);
+      }
+      emailFolder.close(false);
+      store.close();
+    } catch (Exception e) {
+        e.printStackTrace();
+    }
+  }
+
+  
+
+  private Message getMessageById(String id, Message[] messages, IMAPFolder folder){
```

Comment:
Making things private just makes them more difficult to test. Just use protected or package (the default) unless there's a good reason to make private.